### PR TITLE
ttags: init at 0.3.0

### PR DESCRIPTION
**Pull request by mrcjkb**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Output of `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`:

```console
1 package added:
ttags (init at 0.2.1)
1 package built:
ttags
```

### CODE REVIEW

File: pkgs/development/tools/misc/ttags/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      Can be run as a language server that updates the tags
+      for a file when it is saved.
+
+      Supported languages:
+      - JavaScript
+      - Ruby
+      - Rust
+      - Haskell
+    '';
+    homepage = "https://github.com/npezze03/ttags";
```
Comment (techknowlogick) [inline pkgs/development/tools/misc/ttags/default.nix:32]:
```suggestion
    homepage = "https://github.com/npezza93/ttags";
```
there is a minor typo in this link

Comment (mrcjkb) [reply pkgs/development/tools/misc/ttags/default.nix]: Good catch!

**Comment by techknowlogick** [comment]
Result of `nixpkgs-review pr 260722` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ttags</li>
  </ul>
</details>

**Comment by mrcjkb** [comment]
@techknowlogick we're just adding support for Nix and Swift to ttags (in case you want to wait before merging).

File: pkgs/development/tools/misc/ttags/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  version = "0.2.1";
+
+  src = fetchFromGitHub {
+    owner = "npezza93";
+    repo = "ttags";
+    rev = "b8ced4cc71bbf27f514f0084b58e1808a0ea4164";
+    hash = "sha256-nWtsON6VEgXlBao7jxUMIvGJ/8k6cw1I/Zd+aSYzV1I=";
+  };
+
+
```
Comment (techknowlogick) [inline pkgs/development/tools/misc/ttags/default.nix:14]: this file could probably be run through nixfmt, and it'll clean up this double newline and a few other things in this file

**Review by techknowlogick — COMMENTED** [review]
Thanks for this PR, I've made a few comments, feel free to accept or ignore them as you see fit. `ttags` builds and can output the help text and version string on my computer, so from my pov this is okay.

**Comment by techknowlogick** [comment]
@mrcjkb I don't have merge permissions to this repo, just doing my duty of going through the PR list and helping other maintainers with reviews :)

**Comment by teto** [comment]
Result of `nixpkgs-review pr 260722` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ttags</li>
  </ul>
</details>

**Pull request merged**